packages/connect/src/connector/smoldot-light.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import type {
+  Chain as SChain,
+  Client,
+  ClientOptions,
+  AlreadyDestroyedError as IAlreadyDestroyedError,
+  CrashError as ICrashError,
+  JsonRpcDisabledError as IJsonRpcDisabledError,
+} from "@substrate/smoldot-light"
+import {
+  AlreadyDestroyedError,
+  CrashError,
+  JsonRpcDisabledError,
+} from "./errors.js"
+import { getSpec } from "./specs/index.js"
+import type {
+  AddChain,
+  AddWellKnownChain,
+  Chain,
+  SubstrateConnector,
+} from "./types.js"
+import { SupportedChains } from "../SupportedChains.js"
+
+let SdAlreadyDestroyedError: typeof IAlreadyDestroyedError
+let SdCrashError: typeof ICrashError
+let SdJsonRpcDisabledError: typeof IJsonRpcDisabledError
+
+let startPromise: Promise<(options: ClientOptions) => Client> | null = null
+const getStart = () => {
+  if (startPromise) return startPromise
+  return (startPromise = import("@substrate/smoldot-light").then((sm) => {
+    SdJsonRpcDisabledError = sm.JsonRpcDisabledError
+    SdCrashError = sm.CrashError
+    SdAlreadyDestroyedError = sm.AlreadyDestroyedError
```

Comment:
Why is this meta-typing necessary? If it's just to recognize the errors throw by smoldot, can we not use some `name` property or something instead?

Reply:
> If it's just to recognize the errors throw by smoldot, can we not use some name property or something instead?

I wish... Unfortunately, though, [the custom errors created from `smoldot-connect` don't have custom `name` property](https://github.com/paritytech/smoldot/blob/main/bin/wasm-node/javascript/src/client.ts#L22-L53), so that's impossible. So, I'm afraid that this is the only way of recognizing the errors thrown by smoldot, yes.

Reply:
Yeah but I'm implying that we can just modify smoldot. It's not a walled garden.

Reply:
Sure! [here you go](https://github.com/paritytech/smoldot/pull/2019/files). Once, `0.5.21` is published I will update the code accordingly, np!